### DISCUSSION TITLE
closeTransport called before connection was established

### DISCUSSION THREAD

**Discussion by Wsoukkachang** [wevm/wagmi#2574] [Q&A] [answered]
Anybody else getting these errors when trying to use wagmi to connect to WalletConnect? 

- closeTransport called before connection was established

<img width="1153" alt="errors" src="https://github.com/wagmi-dev/wagmi/assets/17499001/9e7e3bba-e6f0-47e1-89b4-ad26269497b7">

- I get it in my dApp and it's showing when trying to use WalletConnect on the wagmi website as well.
[https://wagmi.sh/examples/connect-wallet](url)

<img width="881" alt="wallet" src="https://github.com/wagmi-dev/wagmi/assets/17499001/70f33dfb-e8ee-47df-a37b-8e4b708e0408">

Also in all the demos for v2.

[https://docs.walletconnect.com/2.0/web/guides/examples-and-resources](url)

<img width="1676" alt="demos" src="https://github.com/wagmi-dev/wagmi/assets/17499001/883bd0df-7218-4efa-b446-4648d3e05130">

**Comment by cronosdevl** [comment]
Same error for me too. Possibly this error because project ID of rainbowkit ?
Still, I am facing this error, but sometimes, not happen. but almost.
I am not sure why.

**Comment by kristiegiles** [comment]
Yes, I get this error just from adding the WalletConnect v2 connector.

**Comment by 0xSpiridono** [comment]
same here, annoying

**Comment by Frabal12** [comment]
Same problem here.

**Comment by l3lackcurtains** [comment]
I was able to fix the problem by using connectkit with wagmi.
[https://docs.family.co/connectkit/chains#custom-chains-example](https://docs.family.co/connectkit/chains#custom-chains-example)

**Reply by cronosdevl** [reply]
I was not able to fix this problem by using connectkit too.
I am using following versions.

"@rainbow-me/rainbowkit": "^1.0.2",
"@wagmi/core": "^1.2.0",
"viem": "^1.0.7",
"wagmi": "^1.2.0",

but when I tried your solution, it throws errors like this.

![image](https://github.com/wagmi-dev/wagmi/assets/102214141/0dc03ada-5dfb-4e13-803a-7589fec09359)
:(

**Comment by Holybasil** [comment]
same problem for me

**Comment by Pianochicken** [comment]
same problem here...

**Comment by jxom** [comment]
Replying with “same” isn’t going to help. This is not a wagmi issue. This is a WalletConnect issue. Closing this discussion.

Feel free to open this discussion over on the WalletConnect repo.

**Comment by YuriGii** [comment, marked as answer]
https://github.com/WalletConnect/walletconnect-monorepo/issues/2392

**Reply by jxom** [reply]
Nice find!
